lib/tracker/request-logger.js

```diff
@@ -0,0 +1,281 @@
+ * const requestLogger = new RequestLogger({ slowThreshold: 1000 });
+ * requestLogger.emitter.on('show', message => console.log(message));
+ * // Add the requestLogger to the client options
+ * const client = new Client({ contactPoints, requestTracker: requestLogger });
+ */
+class RequestLogger extends RequestTracker {
+
+  /**
+   * Creates a new instance of {@link RequestLogger}.
+   * @param {Object} options
+   * @param {Number} [options.slowThreshold] The threshold in milliseconds beyond which queries are considered 'slow'
+   * and logged as such by the driver.
+   * @param {Number} [options.requestSizeThreshold] The threshold in bytes beyond which requests are considered 'large'
+   * and logged as such by the driver.
+   * @param {Boolean} [options.logNormalRequests] Determines whether it should emit 'normal' events for every
+   * EXECUTE, QUERY and BATCH request executed successfully, useful only for debugging. This option can be modified
+   * after the client is connected using the property {@link RequestLogger#logNormalRequests}.
+   * @param {Boolean} [options.logErroredRequests] Determines whether it should emit 'failure' events for every
+   * EXECUTE, QUERY and BATCH request execution that resulted in an error. This option can be modified
+   * after the client is connected using the property {@link RequestLogger#logErroredRequests}.
+   * @param {Number} [options.messageMaxQueryLength] The maximum amount of characters that are logged from the query
+   * portion of the message. Defaults to 500.
+   * @param {Number} [options.messageMaxParameterValueLength] The maximum amount of characters of each query parameter
+   * value that will be included in the message.. Defaults to 50.
+   * @param {Number} [options.messageMaxErrorStackTraceLength] The maximum amount of characters of the stack trace
+   * that will be included in the message. Defaults to 200.
+   */
+  constructor(options) {
+    super();
+    if (!options) {
+      throw new errors.ArgumentError('RequestLogger options parameter is required');
+    }
+
+    this._options = options;
+
+    /**
+     * Determines whether it should emit 'normal' events for every EXECUTE, QUERY and BATCH request executed
+     * successfully, useful only for debugging
+     * @type {Boolean}
+     */
+    this.logNormalRequests = this._options.logNormalRequests;
+
+    /**
+     * Determines whether it should emit 'failure' events for every EXECUTE, QUERY and BATCH request execution that
+     * resulted in an error
+     * @type {Boolean}
+     */
+    this.logErroredRequests = this._options.logErroredRequests;
+
+    /**
+     * The object instance that emits <code>'slow'</code>, <code>'large'</code>, <code>'normal'</code> and
+     * <code>'failure'</code> events.
+     * @type {EventEmitter}
+     */
+    this.emitter = new events.EventEmitter();
+  }
+
+  /**
+   * Logs a message when a request execution was deemed slow or too large.
```

Comment:
Shouldn't these comments be:

`onSuccess`:  Logs message if request execution was deemed too slow, large or if normal requests are logged.
`OnError:`  Logs message if request execution was too large and/or encountered an error.

Reply:
Current message makes no sense... I'll fix it!